src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1206,13 +1277,103 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
   // Register field_name in the blueprint_index group.
   if(myid == 0)
   {
-    RegisterFieldInBPIndex(field_name, gf);
+    RegisterFieldInBPIndex(field_name, gf->VectorDim());
```

Comment:
Note that there is a difference between `gf->VectorDim()` and `gf->FESpace()->GetVDim()`. For example, for FE spaces using `mfem::ND_FECollection`, `gf->VectorDim()` will be the same as the `mfem::Mesh::SpaceDimension()` while `gf->FESpace()->GetVDim()` will be 1.

I'm not sure what the intent here is -- I just wanted to point out a potential issue.

Reply:
This logic is mirrored from
https://github.com/mfem/mfem/blob/96fc6f177ea8f2174676e7dc6d5c9ec1a08a4d60/fem/sidredatacollection.cpp#L988-L991

but given the ambiguity I will add that comment to the call site here.